### ISSUE TITLE
Creation of `py.typed` is not detected

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/pyright#7479]
# TL;DR instructions:

```sh
# Install pipx
python -m pip install --user pipx
python -m pipx ensurepath

# Install Poetry
pipx install poetry

# Create a new src-layout project
poetry new pytyped-repro-dir --src

# Enter the environment
cd pytyped-repro-dir
poetry install
poetry shell

# Turn on strict type-checking for Pyright and *disable extra path lookups*.
printf "\n\n[tool.pyright]\n\ntypeCheckingMode = \"strict\"\nextraPaths = []" >> pyproject.toml

# Define a function in the package, and import it from the tests.
printf "def foo():\n    return 'hi'\n" > src/pytyped_repro_dir/__init__.py
printf "from pytyped_repro_dir import foo\n\nfoo()" > tests/__init__.py
```

In a separate terminal, run the following in `pytyped-repro-dir`

```
poetry shell
npm init --yes
npm install pyright
npx pyright --watch
```

Pyright issues the following error:

```
.../pytyped-repro-dir/tests/__init__.py
  .../pytyped-repro-dir/tests/__init__.py:1:6 - error: Import "pytyped_repro_dir" could not be resolved (reportMissingImports)
  .../pytyped-repro-dir/tests/__init__.py:1:31 - error: Type of "foo" is unknown (reportUnknownVariableType)
```

In the original terminal, create a `py.typed` file:

```
touch src/pytyped_repro_dir/py.typed
```

Notice file watching doesn't pick up any change. The error persists, but goes away if you restart Pyright or the language service.
_______

# Long story

Note, most of the contents here are at https://github.com/DanielRosenwasser/daniel-ts-repros/tree/pytyped-file

Create a `src`-layout package like the following in poetry by running `poetry new --src pytyped-repro-dir`

```
.
├── pyproject.toml
├── src
│   └── pytyped_repro_dir
│       └── __init__.py
└── tests
    └── __init__.py
```

Ensure you have the following contents:

**`src/pytyped_repro_dir/__init__.py`**

```
def foo():
    return "hi"
```

**`tests/__init__.py`**

```
from pytyped_repro_dir import foo

foo()
```

Install and enter a shell.

```
poetry install
poetry shell
```

Set the following options for Pyright in `pyproject.toml` or `pyrightconfig.json`

```
typeCheckingMode = "strict"
extraPaths = []
```

Run pyright in watch mode

```
npx pyright --watch
```

At this point, you should see an error like the following:

```
.../pytyped-repro-dir/tests/__init__.py
  .../pytyped-repro-dir/tests/__init__.py:1:6 - error: Import "pytyped_repro_dir" could not be resolved (reportMissingImports)
  .../pytyped-repro-dir/tests/__init__.py:1:31 - error: Type of "foo" is unknown (reportUnknownVariableType)
```

Now create a `py.typed` file in `src/pytyped_repro_dir`

```sh
touch src/pytyped_repro_dir/py.typed
```

Notice that the error persists and a new check isn't triggered.

**Comment by erictraut** [comment]
I don't think this is worth fixing. I'm guessing that you're the first person who has ever hit this.

If the pylance team would like to make it a priority and have a way to fix it without overly complicating the code or causing other harm in the code base, then they're welcome to do so. 

I'm also willing to revisit this if there's signal from other pyright users that this is a real issue, but for now I'm going to close this.

**Comment by DanielRosenwasser** [comment]
> I'm guessing that you're the first person who has ever hit this.

Look, I get that I'm hitting the edge cases a lot... 😄

Searching on Google for the text:

```
Stub file not found for
```

a top result is this comment from 2021 https://github.com/mkorpela/overrides/issues/90#issuecomment-920674807

> Probably pylance has not updated after the update. pylance restart was not enough for some reason.

As a developer trying to configure a repository with strict checking, this is a weird error-message - everything in my codebase can be fully typed, but I'm told that there are no stub files. It doesn't mention `py.typed` at all. However, when I search around for a bit, I realize that I need to add a `py.typed` file. When I do that, the editor doesn't pick up on it!

Honestly, the likeliest thing is that developers are restarting and shrugging it off, closing their window and trying again the next day, or possibly giving up on stricter type-checking entirely.
